src/arcaflow_plugin_sdk/schema.py

```diff
@@ -2807,6 +2811,9 @@ class OneOfStringSchema(OneOfSchema):
     oneof_type: typing.Annotated[str, _name("One Of String Type Name")] = (
         "_discriminated_string_"
     )
+    discriminator_type: typing.Annotated[str, _name("Discriminator Type")] = (
+        "string"
+    )
```

Comment:
Won't do. Those should work b/c I iterated through this solution at one point, but @dbutenhof comment on the previous review on these constructors encouraged me to get to where they are now. 

Don't you think generics are a big hammer for this problem (=?

Reply:
> Don't you think generics are a big hammer for this problem (=?

I think that this is a different problem from the one that you were trying to solve with generics.  😁 

> Won't do. Those should work b/c I iterated through this solution at one point, but @dbutenhof comment on the previous review on these constructors encouraged me to code this current implementation.

_In general_, one should avoid defining `__init__()` functions for `@dataclass`'s, because the decorator already does that for you, and it likely does a better job.  The problem here is, we're creating a subclass, and we're overriding the parent class's _attributes_ (which, really, is kind of weird), and, worse, in two of the cases we're doing it just so that we can specify default values, not because we want to change their types.  (Arguably, in the third case, `types`, the base class should not be providing the attribute at all, since the subclasses always override it...but the base class member functions reference it, which makes it awkward for it to be undefined, but only in Python could we have a reference to something that we didn't define.... 😛)

So, I think we're left to choose between evils:  either we override the attributes so that we can add default values to them, or we put a hook into the initialization of them so that we can provide default values that way.

I think the subclasses _should_ override the `types` attribute, so that they can set the key type appropriately; however, I think the subclasses _should not_ override the other two attributes, because they have no compelling reason to do so.  We should instead be providing the required initializations either through a custom `__init__()` function or via a `__post_init__()` function.  Given the actual use, I believe that these attributes always have the same values, so there is no need for them to be set by the calling code, which is, in fact, the impulse for adding the default values to the class definitions.

So, yeah, we should follow the practice of not providing our own `__init__()` function, but we should go on from there and make it explicit that the values of `oneof_type` and `discriminator_type` cannot be set by the creating caller, and instead set them via a `__post_init__()` function.  I think that does all the right things.  (The reason that I didn't propose it initially was a desire to retain the ability for the caller to set those fields on construction, but, in retrospect, that's not really a reasonable thing to do, we don't do it currently, and, by removing that constraint, we can arrive a good solution.)

Reply:
now it is factored out into `__post_init__()`